spec/requests/api/moves_controller_index_v2_spec.rb

```diff
@@ -0,0 +1,235 @@
+          data: [{ id: move.id }],
+        )
+      end
+    end
+
+    describe 'paginating results' do
+      let!(:moves) { create_list :move, 6 }
+
+      let(:meta_pagination) do
+        {
+          per_page: 5,
+          total_pages: 2,
+          total_objects: 6,
+          links: {
+            first: '/api/moves?page=1',
+            last: '/api/moves?page=2',
+            next: '/api/moves?page=2',
+          },
+        }
+      end
+
+      before { do_get }
+
+      it_behaves_like 'an endpoint that paginates resources'
+    end
+
+    describe 'validating dates before running queries' do
+      let(:from_location) { moves.first.from_location }
+      let(:filters) do
+        {
+          from_location_id: from_location.id,
+          date_from: 'yyyy-09-Tu',
+        }
+      end
+      let(:params) { { filter: filters } }
+
+      before { do_get }
+
+      it_behaves_like 'an endpoint that responds with error 422' do
+        let(:errors_422) do
+          [{ 'title' => 'Invalid date_from',
+             'detail' => 'Validation failed: Date from is not a valid date.' }]
+        end
+      end
+    end
+
+    describe 'included relationships' do
+      let!(:moves) do
+        create_list(
+          :move,
+          1,
+          from_location: from_location,
+          to_location: to_location,
+        )
+      end
+
+      let!(:court_hearing) { create(:court_hearing, move: moves.first) }
+
+      let(:to_location) { create(:location, suppliers: [supplier]) }
+      let(:from_location) { create(:location, suppliers: [supplier]) }
+
+      before do
+        get "/api/moves#{query_params}", params: params, headers: headers
+      end
+
+      context 'when not including the include query param' do
+        let(:query_params) { '' }
+
+        it 'returns the default includes' do
+          returned_types = response_json['included']
+          expect(returned_types).to be_nil
+        end
+      end
+
+      context 'when including the include query param' do
+        let(:query_params) { '?include=profile' }
+
+        it 'includes the requested includes in the response' do
+          returned_types = response_json['included'].map { |r| r['type'] }.uniq
+          expect(returned_types).to contain_exactly('profiles')
+        end
+      end
+
+      context 'when including an invalid include query param' do
+        let(:query_params) { '?include=foo.bar,profile' }
+
+        let(:expected_error) do
+          {
+            'errors' => [
+              {
+                'detail' => match(/foo.bar/),
+                'title' => 'Bad request',
+              },
+            ],
+          }
+        end
+
+        it 'returns a validation error' do
+          expect(response).to have_http_status(:bad_request)
+          expect(response_json).to include(expected_error)
+        end
+      end
+    end
+  end
+
+  context 'when not authorized', :skip_before, :with_invalid_auth_headers do
```

Comment:
I thought we removed `skip_before` for the v2 tests

Reply:
Does nothing so can remove

Reply:
```suggestion
  context 'when not authorized' do
```